Debug: Add Hello World Gradle plugin with auto-execution during app build [#2, #4]

pakagronglb opened the pull request:
Hopefully this would work this time. I tried debugging and testing multiple times, although it took a while (probably due to my computer). Too many commits lol but it was so damn difficult yet fun to tinker. 

Changes Made

1. Added a custom Gradle plugin in buildSrc directory that:
   - Adds `hello` and `projectInfo` tasks to display basic project information
   - Automatically executes during the app build process without manual invocation
   - Shows proper log messages in build output

2. Removed buildSrc/build directory (generated files) from repository
   - Added .gitignore rule to prevent future commits of build artifacts
   - Kept only source files to ensure clean repo structure

3. Made plugin tasks integrate with Android build system:
   - Added the hello task as a dependency to preBuild task
   - Used proper lifecycle logging for visibility in build output

Testing

1. Plugin tasks now show up in build logs when running `./gradlew :app:assembleDebug`
2. Tasks run automatically as part of the build process
3. No need to manually invoke `./gradlew hello` anymore

<img width="626" alt="Screenshot 2025-03-29 at 07 18 13" src="https://github.com/user-attachments/assets/4c33d19f-d473-4ad3-9031-1348d006b6fe" />

Hopefully this aligns with your vision or at least closer to what you're hoping for. If not, feel free to let me know!

mcallisto commented:
> Hopefully this aligns with your vision or at least closer to what you're hoping for. If not, feel free to let me know!

Hi @pakagronglb, this is excellent and completely solves our #2! Thank you!

What I would ask you is to delete the files:
* `.vscode/settings.json`
* `core/bin/main/com/example/core/Foo.scala`

You can do that in your latest `scala-in-app-module` branch, commit the change and it should automatically appear in this PR.

After that I will be able to merge your contribution.

P.S. In the meanwhile I will open a new issue trying to outline what is expected from the plugin compiling Scala code found in the `app` module.

pakagronglb commented:
> > Hopefully this aligns with your vision or at least closer to what you're hoping for. If not, feel free to let me know!
> 
> Hi @pakagronglb, this is excellent and completely solves our #2! Thank you!
> 
> What I would ask you is to delete the files:
> 
> * `.vscode/settings.json`
> * `core/bin/main/com/example/core/Foo.scala`
> 
> You can do that in your latest `scala-in-app-module` branch, commit the change and it should automatically appear in this PR.
> 
> After that I will be able to merge your contribution.
> 
> P.S. In the meanwhile I will open a new issue trying to outline what is expected from the plugin compiling Scala code found in the `app` module.

@mcallisto I think i did but the commit files didnt go through. I'll do it again now. Sorry about that. My mistake!

pakagronglb commented:
I've completed all the requested tasks and have grown a love-hate relationship with my github desktop version. Sorry for the delay and constant debugging. I was hoping things were smoother. Anyways I:

1. Deleted both files:
   - `.vscode/settings.json`
   - `core/bin/main/com/example/core/Foo.scala`

2. Added both directories to `.gitignore` (as shown in the attached `.gitignore` file)

3. Committed and pushed these changes to the `scala-in-app-module` branch

The commit that removed these files is already in the PR (commit 9ed889c), and the `.gitignore` has been updated to prevent these files from being committed in the future. Your PR should now be ready for merging. If I did anything right or wrong just let me know, happy to help regardless. I know im still a beginner but I'm willing to learn from feedback.

mcallisto commented:
@pakagronglb very good, thank you again. See now #6, if you are interested in going further!

pakagronglb commented:
> @pakagronglb very good, thank you again. See now #6, if you are interested in going further!

Absolutely!! Thank you so much for the opportunity :D
